CONTRIBUTING.md

```diff
@@ -31,17 +31,23 @@ All developers including new distributions and plugin developers are invited to
 
 ### How to submit PR's (pull requests), patches, and bug fixes
 
-- Avoid merging master in your branch because it makes code review a lot harder.
-- If you want to keep your branch up to date with our master, it would be nicer if you could just rebase your branch instead. That would keep the history a lot cleaner.
-- Please avoid adding unrelated changes in the PR. Do a separate PR and rebase once they get merged can work really well.
- 
-If you make trivial changes, you can send them directly via a pull request. **Please make your changes in a new git branch and send your patch**, including appropriate test cases.
+Welcome new contributor!
+Follow this guide [Your first pull request](https://gitter.im/OpenRefine/OpenRefine)
 
-We want to keep the quality of the trunk at a very high level, since this is ultimately where the Stable Releases are built from after bugs are fixed. Please take the time to test your changes (including travis-ci) before sending a pull request.
+- Avoid merging master in your branch because it makes code review a lot harder. **Please make your changes in a new git branch and send your patch**, including appropriate test cases.
+- If you want to keep your branch up to date with our master, it would be nicer if you could just rebase your branch instead. That would keep the history a lot cleaner.
+- Please avoid adding unrelated changes in a PR. Use separate PR's and rebase once they get merged can work really well.
+- If you make trivial changes, you can send them directly via a PR.
 
-OpenRefine is volunteer supported. Pull Requests are reviewed and merged by volunteers. All Pull Requests will be answered, however it may take some time to get back to you. Thank you in advance for your patience.
+### Code Quality ###
+We want to keep the quality of the trunk at a very high level, since this is ultimately where the Stable Releases are built from after bugs are fixed.
+- For Java code, we follow **Google Java Style** (Use the [Eclipse Plugin](https://github.com/google/google-java-format#eclipse)).
+- For JavaScript, we follow **JavaScript Standard Style** (Eclipse Orion and greater [defaults to ESLint](https://eslint.org/docs/3.0.0/user-guide/integrations)).
```

Comment:
Shouldn't we discuss that first? (or have these styles been chosen before? Any links?)

When we make that decision it would be good to check if we can add linters to check for issues in pull requests (our choice of style might depend on the availably of these tools).

Reply:
That's what we are here for...to discuss!  Oops!  forgot to make this as DRAFT!

Reply:
ah crap, can't switch back to draft it seems.

Reply:
The discussion happened long ago, just a few basics.  But generally for Java it was Google Code Style more or less. https://groups.google.com/forum/#!searchin/openrefine/code$20style%7Csort:date/openrefine/l7iiTb7vYgs/Lf-kYDOkn5oJ

but nothing historical for Javascript other than the Jquery recommendation at the time (pre-ECMA stuff), so David followed that.

Reply:
ah, and this commit and a few others later...  https://groups.google.com/forum/#!searchin/openrefine-dev/style%7Csort:date/openrefine-dev/THkElQ_4W-Q/QnRRnORvAnIJ

Reply:
Sorry, here's the real one...  https://groups.google.com/forum/#!searchin/openrefine-dev/style%7Csort:date/openrefine-dev/F4KJsMlEU3I/gLfnRdVkwDYJ